File: asp.net-projects/Maps_Project/Maps_Project.Web/Modules/ServerTypes/Trip/BusinessTripForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface BusinessTripForm {
    Title: StringEditor;
    Description: StringEditor;
    DetailList: StringEditor;
}

export class BusinessTripForm extends PrefixedContext {
    static readonly formKey = 'Trip.BusinessTrip';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BusinessTripForm.init)  {
            BusinessTripForm.init = true;

            var w0 = StringEditor;

            initFormType(BusinessTripForm, [
                'Title', w0,
                'Description', w0,
                'DetailList', w0
            ]);
        }
    }
}

File: asp.net-projects/Maps_Project/Maps_Project.Web/Modules/Trip/BusinessTrip/BusinessTripForm.cs
using Serenity.ComponentModel;

namespace Maps_Project.Trip.Forms;

[FormScript("Trip.BusinessTrip")]
[BasedOnRow(typeof(BusinessTripRow), CheckNames = true)]
public class BusinessTripForm
{
    public string Title { get; set; }
    public string Description { get; set; }


    [BusinessTripDetail]
    public List <BusinessTripDetailRow> DetailList { get; set; }



}